portal-web/test/functional/com/liferay/portalweb/macros/ClientExtension.macro

```diff
@@ -0,0 +1,60 @@
+definition {
+
+	macro addPortletToWidgetPage {
```

Comment:
Good to learn, and just a small tip that as far as I know the `JsonLayout.updateLayputTemplateOfPublicLayout();` is not necessary for all cases. I mean `JSONLayout.addPublicLayout()` and `JSONLayout.addWidgetToPublicLayout()` are enough. :D

Reply:
I was using the adding because of the issue we had when placing a widget on the page (it did not render well), so it's to make sure that it does do it immediately after the addition of the custom element created through UI. If we consider it not important in this test I can change it to JSON.

Reply:
icic, in this case, I'd suggest making a new separate testcase for that particular issue.

I'm trying not to overlap or add too many dependencies in one testcase, unless the goal of the test is a usecase/smoke test. The goal we're aiming for is that when tests modeled after SACRED fail, we can more immediately know with certainty what failed and not have to investigate as much inside the test.

Reply:
yeah, I was thinking about it, but having in mind the CI problem, and that this is "easy" to wrap up in one test maybe it is easier to leave it. Otherwise, we need to add another test - Assertion in When would be the same for both tests in this case

Reply:
Let's discuss more in our sync tomorrow to better understand.

Reply:
@john-co I have changed the title of the test and one of the steps so I hope it is more clear now :) If you'd like to change anything in the copy, feel free, if not, kindly please to forward :) Thanks for all the suggestions!